client/src/Pos/Client/Txp/Util.hs

```diff
@@ -222,46 +229,49 @@ runTxCreator inputSelectionPolicy action = runExceptT $ do
 
 -- | Like 'makePubKeyTx', but allows usage of different signers
 makeMPubKeyTx
-    :: HasConfiguration
-    => (owner -> SafeSigner)
+    :: (HasConfiguration)
+    => (owner -> Either e SafeSigner)
     -> TxOwnedInputs owner
     -> TxOutputs
-    -> TxAux
+    -> Either e TxAux
 makeMPubKeyTx getSs = makeAbstractTx mkWit
   where mkWit addr sigData =
-          let ss = getSs addr
-          in PkWitness
+          getSs addr <&> \ss ->
+              PkWitness
               { twKey = safeToPublic ss
               , twSig = safeSign SignTx ss sigData
               }
 
 -- | More specific version of 'makeMPubKeyTx' for convenience
 makeMPubKeyTxAddrs
-    :: HasConfiguration
-    => (Address -> SafeSigner)
+    :: (HasConfiguration)
+    => (Address -> Either e SafeSigner)
     -> TxOwnedInputs TxOut
     -> TxOutputs
-    -> TxAux
+    -> Either e TxAux
 makeMPubKeyTxAddrs hdwSigners = makeMPubKeyTx getSigner
   where
     getSigner (TxOut addr _) = hdwSigners addr
 
 -- | Makes a transaction which use P2PKH addresses as a source
 makePubKeyTx :: HasConfiguration => SafeSigner -> TxInputs -> TxOutputs -> TxAux
-makePubKeyTx ss txInputs =
-    makeMPubKeyTx (const ss) (map ((), ) txInputs)
+makePubKeyTx ss txInputs txOutputs = either absurd identity $
```

Comment:
Is this really what we want? Can `makeMPubKeyTx` return `Left`? Wouldn't a pattern matching be more explicit here? For once, it would force us to think about how to handle the `Left` case of the branch.

Reply:
There's no `Left` case of the branch here, it's absurd.

Reply:
It's an unnecessarily fancy trick. Not having to have Either in the first place (by abstracting over Monad m) is a good deal simpler and more familiar to readers.

Reply:
Excuse me @dcoutts, but I discussed this with @pva701 and my understanding was that initially the plan was to make `getSigner` monadic, then *you* insisted that it should not be monadic. A previous version of this PR was using the `m` monad here, but then I was made aware that you disapprove that and I changed it to `Either`.

I do not intend to refactor again, the current version works fine and the types are more precise.